src/js/with-react.js

```diff
@@ -59,19 +61,36 @@ const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
       super(props);
 
       this._eventCache = {};
+
+      this.state = {
+        isDefined: false,
+      };
+
+      // make sure that the custom element is ready
+      // ref: https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/whenDefined
+      customElements.whenDefined(tagName).then(() => {
```

Comment:
IE11 does not have `Promise`

<img width="1438" alt="screen shot 2018-07-05 at 13 14 16" src="https://user-images.githubusercontent.com/914443/42320329-4687924e-8055-11e8-9286-478a00415c22.png">